providers/structured/provider.go

```diff
@@ -34,12 +35,21 @@ type StructuredOutputLogProvider struct {
 	rawLogCalls      map[providers.RawLogCallType][]RawLogCallArgs
 	recordCalls      []RecordCallArgs
 	recordEventCalls []RecordEventCallArgs
+	logMutex         sync.RWMutex
```

Comment:
What’s the reason that this an RWMutex but the others are Mutex?

Reply:
Don't want multiple readers blocking each other. A little more performant than a Mutex in case you only have readers.

Reply:
OIC, that makes sense.